AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyAssetEditor.py

```diff
@@ -0,0 +1,163 @@
+        Refer to github issue #12262. If it has been resolved and this comment still exists please update
+        the QToolButton name
+        """
+        methods = self.asset_editor_row_container.findChildren(QtWidgets.QFrame, DEFAULT_SCRIPT_EVENT)
+        for method in methods:
+            if method.findChild(QtWidgets.QToolButton, ""):
+                method.findChild(QtWidgets.QToolButton, "").click()
+
+    def save_script_event_file(self, file_path: str) -> None:
+        """
+        Function for saving the currently open script event file to disk then verifying the save action occurred
+
+        param file_path: The path on disk to save the file to
+
+        returns: None
+        """
+        editor.AssetEditorWidgetRequestsBus(bus.Broadcast, "SaveAssetAs", file_path)
+        action = pyside_utils.find_child_by_pattern(self.menu_bar,
+                                                    {"type": QtWidgets.QAction, "iconText": SAVE_STRING})
+
+        assert action is not None, "Unable to complete save action. Filepath may be invalid."
+
+        action.trigger()
+        self.__verify_save_action(file_path)
+
+    def __verify_save_action(self, file_path: str) -> None:
+        """
+        helper function to make the save_script_event_file function a bit cleaner. Performs verification that
+        the save action occurred by checking the tab's label for the * and checking the location on disk for a file
+
+        param file_path: The location of the created file on disk
+
+        returns: None
+        """
+        label = self.asset_editor.findChild(QtWidgets.QLabel, "textEdit")
+        saved = helper.wait_for_condition(lambda: "*" not in label.text(), WAIT_TIME_SEC_3)
+        exists = helper.wait_for_condition(lambda: os.path.exists(file_path), WAIT_TIME_SEC_3)
+
+        assert saved, "Save file failed. Save action not detected in UI (* in label)."
+        assert exists, "Save file failed. File not located on disk."
+
+    def expand_category_by_name(self, category_name: str) -> None:
+        """
+        Note: I think I want to move this to the Common class and make the container you want to explore a parameter
+        Function for expanding a category container in a list.
+
+        param category_name: the category in the list you want to expand.
+
+        returns: None
+        """
+        children = self.asset_editor_row_container.findChildren(QtWidgets.QFrame, category_name)
+        assert children is not None, f"Category row by name {category_name} was not found."
+        for child in children:
+            check_box = child.findChild(QtWidgets.QCheckBox)
+            if check_box and not check_box.isChecked():
+                check_box.click()
+
+    def gather_row_container_types(self):
+        children = self.asset_editor_row_container.children()
+        depth = 0
+        for child in children:
+            print(f"{depth}->Name:{child.objectName()}: {type(child)}")
```

Comment:
Remove debug text.

Reply:
removed this function since it was strictly for debug printing the structure of qt widgets